zerver/signals.py

```diff
@@ -13,51 +13,34 @@
 from zerver.lib.send_email import FromAddress
 from zerver.models import UserProfile
 from zerver.lib.timezone import get_timezone
+from ua_parser import user_agent_parser
 
 JUST_CREATED_THRESHOLD = 60
 
 def get_device_browser(user_agent: str) -> Optional[str]:
-    user_agent = user_agent.lower()
-    if "zulip" in user_agent:
+    device_browser = user_agent_parser.ParseUserAgent(user_agent)['family']
+    if "zulip" in user_agent.lower():
         return "Zulip"
-    elif "edge" in user_agent:
-        return "Edge"
-    elif "opera" in user_agent or "opr/" in user_agent:
-        return "Opera"
-    elif ("chrome" in user_agent or "crios" in user_agent) and "chromium" not in user_agent:
-        return 'Chrome'
-    elif "firefox" in user_agent and "seamonkey" not in user_agent and "chrome" not in user_agent:
-        return "Firefox"
-    elif "chromium" in user_agent:
-        return "Chromium"
-    elif "safari" in user_agent and "chrome" not in user_agent and "chromium" not in user_agent:
-        return "Safari"
-    elif "msie" in user_agent or "trident" in user_agent:
+    elif device_browser == "IE":
         return "Internet Explorer"
+    elif device_browser == "Chrome Mobile iOS":
+        return "Chrome"
+    elif device_browser == "Mobile Safari":
+        return "Safari"
```

Comment:
I think we should change the tests to match the unmodified `ua_parser` return value, instead of the other way around. Outside of tests, we only use this as a human-readable string in the `notify_new_login` template.

Reply:
In the email it would look like "Device: IE on windows", "Device: Chrome Mobile iOS on iOS", "Device: Mobile Safari on iOS" with the change, which feels a little odd to me.  Do you feel it's alright? I'm not sure about this. Shall I change tests and commit?

Reply:
Chrome Mobile iOS probably *should* be treated as a different browser than Chrome, since it uses the mobile Safari rendering engine, not the Chrome engine. In general I would prefer to delegate these questions to the library. If the display feels a little odd but we never have to deal with these issues in our code, that seems like a fine tradeoff to me.

Reply:
>Chrome Mobile iOS probably should be treated as a different browser than Chrome, since it uses the mobile Safari rendering engine, not the Chrome engine

That is probably the reason why `ua_parser` chose to return "Chrome Mobile iOS" but in the email, we also mention the OS(iOS) which I think will make it seem like different browser than Chrome.

> If the display feels a little odd but we never have to deal with these issues in our code, that seems like a fine tradeoff to me.

Okay, I think it's better to change it to match with the unmodified `ua_parser` return value as there's a chance for those strings(return values) to be modified by the library later. I'm curious to know what @timabbott thinks, before changing.

Reply:
Yeah, I agree, I'd rather see "Mobile Safari" than "Safari" if it's indeed Safari on mobile